yls/utils.py

```diff
@@ -579,3 +670,87 @@ async def start_progress(ls: Any, msg: str) -> str | None:
         )
 
     return token
+
+
+def extract_rule_context_from_yarafile(yara_file: yaramod.YaraFile, rule: yaramod.Rule) -> str:
+    """Extract specified rule, private rules and modules it is dependent on from YaraFile."""
+    # Find private rule references inside the condition
+    all_rules = yara_file.rules
+
+    def _extract(curr_rule: yaramod.Rule, extracted_rules: List[str]) -> List[str]:
```

Comment:
Nitpick: Use new style of typehints for consistency
```suggestion
    def _extract(curr_rule: yaramod.Rule, extracted_rules: list[str]) -> list[str]:
```